llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -3248,12 +3249,18 @@ bool SelectionDAGLegalize::ExpandNode(SDNode *Node) {
       Results.push_back(Tmp1);
     break;
   }
+  case ISD::STRICT_BF16_TO_FP:
```

Comment:
We cannot support `STRICT_BF16_TO_FP` by `SHL` because it cannot generate expected exceptions.

Reply:
This is more like a "fallback" to the non-strict variant when the strict FP is not enforced, which is the purpose of the check below. We have something similar in this big `switch`, such as `case ISD::STRICT_FP_EXTEND`.

Reply:
They are different. The comment says the expansion is not allowed when strict mode is enforced. Here you just do the expansion unconditionally.

Reply:
Yeah, I did have a check earlier but removed it since I think https://github.com/llvm/llvm-project/pull/80056#discussion_r1472611861 makes sense.